frontend/src/container/QueryBuilder/QueryBuilder.tsx

```diff
@@ -33,6 +34,13 @@ export const QueryBuilder = memo(function QueryBuilder({
 		};
 	}, [config, setupInitialDataSource]);
 
+	useEffect(
```

Comment:
can you check why there is a need to create a new use effect to do the reset?

https://react.dev/learn/removing-effect-dependencies#read-primitive-values-from-objects